llvm/utils/TableGen/Basic/CodeGenIntrinsics.cpp

```diff
@@ -94,6 +95,168 @@ void CodeGenIntrinsicTable::CheckDuplicateIntrinsics() const {
   PrintFatalNote(First.TheDef, "Previous definition here");
 }
 
+// Note: This is a modified version of `Intrinsic::lookupLLVMIntrinsicByName`
+// in IntrinsicInst.cpp file.
+template <typename T>
```

Comment:
1st point of feedback: I made this templated so that we can share the same code here and in `Intrinsic::lookupLLVMIntrinsicByName`. I am proposing we move this to a common header in Support that both pieces of code use.